### PR TITLE
modifies IndexingJob (and adds DeleteFromAllIndexesJob) to work with …

### LINKED ISSUE #77: Update to work with base-indexer v3.0 api
See: https://github.com/sul-dlss/base_indexer/blob/cbacb81c83951d3a8037580ddc4c09a2aa733e2b/README.md#http-api

### PR DESCRIPTION
**Pull request by mejackreed**
…the BaseIndexer v3.0 api changes

Closes #77 

Works with the new changes added in https://github.com/sul-dlss/base_indexer/pull/34

### CODE REVIEW

File: app/models/purl_fetcher/record_deletes.rb [hunk 1]
```diff
@@ -11,8 +11,8 @@ def enqueue
         ##
         # The `target` here is an Array of configuration values, the first value
         # being the name of the target.
-        Delayed::Job.enqueue(  
-          DiscoveryDispatcher::IndexingJob.new('delete', druid, target[0].to_s.downcase)
+        Delayed::Job.enqueue(
+          DiscoveryDispatcher::DeleteFromAllIndexesJob.new('delete', druid, target[0].to_s.downcase)
```
Comment (drh-stanford) [inline app/models/purl_fetcher/record_deletes.rb:15]: shouldn't `target[0].to_s.downcase` just be `nil`?

File: lib/discovery_dispatcher/indexing_job.rb [hunk 2]
```diff
@@ -20,13 +21,15 @@ def perform
...
+    # By convention we are using settings on sw-indexer that specify subtargets
+    # in all caps (thus the target upcase).
     # @param druid [String] represents the druid on the form of ab123cd4567
     # @param target_url [String] the url for the indexing service
     # @param solr_target [Array] solr target for solr core
     # @return [String] RestClient request command
     def build_request_url(druid, target_url, solr_target)
-      solr_target = { solr_target.upcase => true }
-      "#{target_url}/items/#{druid}?#{solr_target.to_query('subtargets')}"
+      "#{target_url}/items/#{druid}/subtargets/#{solr_target.upcase}"
```
Comment (drh-stanford) [inline lib/discovery_dispatcher/indexing_job.rb:32]: the param doc ^^^ says `solr_target` is an Array. the `.upcase` assumes it's a String. should it be `first.to_s.upcase` instead? or is the doc wrong

File: spec/models/purl_fetcher/record_deletes_spec.rb [hunk 3]
```diff
@@ -13,8 +13,8 @@
         described_class.new(druid: 'abc123')
       end
       it 'enqueue a delete job' do
-        expect(Delayed::Job).to receive(:enqueue).with(DiscoveryDispatcher::IndexingJob.new('delete', 'abc123', 'searchworkspreview'))
-        expect(Delayed::Job).to receive(:enqueue).with(DiscoveryDispatcher::IndexingJob.new('delete', 'abc123', 'searchworks'))
+        expect(Delayed::Job).to receive(:enqueue).with(DiscoveryDispatcher::DeleteFromAllIndexesJob.new('delete', 'abc123', 'searchworkspreview'))
+        expect(Delayed::Job).to receive(:enqueue).with(DiscoveryDispatcher::DeleteFromAllIndexesJob.new('delete', 'abc123', 'searchworks'))
         subject.enqueue
```
Comment (drh-stanford) [inline spec/models/purl_fetcher/record_deletes_spec.rb:18]: i find the name `DeleteFromAll...` combined with the subtarget confusing. i would expect a single expect for the DeleteFromAll case (i.e., no subtargets).

File: spec/discovery_dispatcher/delete_from_all_indexes_job_spec.rb [hunk 4]
```diff
@@ -0,0 +1,37 @@
+require 'rails_helper'
+
+describe DiscoveryDispatcher::DeleteFromAllIndexesJob do
+  describe '.perform' do
+    it 'performs the indexing process successfully' do
+      stub_request(:delete, 'http://www.example-indexer.com/items/xz404nk7341').
+        to_return(status: 200)
+      index_job = described_class.new('delete', 'xz404nk7341', 'searchworkspreview')
+      expect(Rails.logger).to receive(:debug).with(/Processing/).and_call_original
```
Comment (drh-stanford) [inline spec/discovery_dispatcher/delete_from_all_indexes_job_spec.rb:9]: i find it confusing that you're passing in `searchworkspreview` to a delete all case.

**Comment by lmcglohon** [comment]
sorry @drh-stanford didn't realize you were looking at this one...

File: lib/discovery_dispatcher/indexing_job.rb (hunk 2 above)
Comment (lmcglohon) [inline lib/discovery_dispatcher/indexing_job.rb:32]: I think the doc is wrong @drh-stanford

File: app/models/purl_fetcher/record_deletes.rb (hunk 1 above)
Comment (mejackreed) [inline app/models/purl_fetcher/record_deletes.rb:15]: Could be yeah.

File: spec/models/purl_fetcher/record_deletes_spec.rb (hunk 3 above)
Comment (mejackreed) [inline spec/models/purl_fetcher/record_deletes_spec.rb:18]: Definitely a fair assessment here. I'll work on cleaning this up.

Comment (mejackreed) [inline spec/models/purl_fetcher/record_deletes_spec.rb:18]: I think that this information is still needed. Unless we modify the `IndexingJob` to take a URL param, we still need to know how to lookup which indexer to send the request to. It definately is confusing, but the "delete from all" really happens in the `RecordDeletes#enqueue` method.

**Pull request merged**